src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -361,6 +361,10 @@ public function getProjectDir()
      */
     public function getContainer()
     {
+        if (null === $this->container) {
+            @trigger_error('Getting the container from a non-booted kernel is deprecated since 4.3.', E_USER_DEPRECATED);
```

Comment:
since Symfony 4.3.

Reply:
The `KernelInterface` interface documents the null return value, which should be modified as well. I think we should also add a hint about the fact that this method will return an exception in 5.0 like we do elsewhere.